src/test/ui/abi/stack-probes.rs

```diff
@@ -50,8 +50,7 @@ fn main() {
 #[allow(unconditional_recursion)]
 fn recurse(array: &[u64]) {
     unsafe { black_box(array.as_ptr() as u64); }
-    #[allow(deprecated)]
-    let local: [_; 1024] = unsafe { mem::uninitialized() };
+    let local: [u64; 1024] = [0; 1024];
```

Comment:
Is zero-initializing the buffer here going to defeat the purpose of the test?

Reply:
I don't think so. This test (as best as I can tell) is making sure that we properly hit the end of the stack, and the kilobyte large array just makes sure that we get there faster.

Reply:
Note that the array is actually 8KB. This test was added in #42816 to test stack probing - which is only necessary when the stack frame might jump over the guard page - whereas theoretically the memset inserted here could trigger the guard page on its own. Though it probably wouldn't because it would probably write in the wrong order for that.

Reply:
The old intent could be preserved using `MaybeUninit<[u64; 1024]>` and `*const [u64; 1024]`, I think.

Reply:
Ah, I see what you mean. I will change this to MaybeUninit then, good catch!